core/modules/qproc/IndexMap.cc

```diff
@@ -0,0 +1,252 @@
+#include "boost/lexical_cast.hpp"
+
+// LSST headers
+#include "sg/Chunker.h"
+
+// Qserv headers
+#include "global/intTypes.h"
+#include "global/stringTypes.h"
+#include "qproc/geomAdapter.h"
+#include "qproc/SecondaryIndex.h"
+#include "query/Constraint.h"
+
+using lsst::qserv::StringVector;
+using lsst::sg::Region;
+using lsst::sg::Box;
+using lsst::sg::Circle;
+using lsst::sg::Ellipse;
+using lsst::sg::ConvexPolygon;
+using lsst::sg::SubChunks;
+using lsst::qserv::qproc::Coordinate;
+
+typedef std::vector<SubChunks> SubChunksVector;
+
+namespace { // File-scope helpers
+template <typename T>
+std::vector<T> convertVec(StringVector const& v) {
+    std::vector<T> out;
+    std::transform(v.begin(), v.end(), std::back_inserter(out),
+                   boost::lexical_cast<T, std::string>);
+    return out;
+}
+template <typename T>
+boost::shared_ptr<Region> make(StringVector const& v) {
+        return boost::shared_ptr<Region>(new T(convertVec<Coordinate>(v)));
+}
+template <>
+boost::shared_ptr<Region> make<Box>(StringVector const& v) {
+    return lsst::qserv::qproc::getBoxFromParams(convertVec<Coordinate>(v));
+}
+template <>
+boost::shared_ptr<Region> make<Circle>(StringVector const& v) {
+    return lsst::qserv::qproc::getCircleFromParams(convertVec<Coordinate>(v));
+}
+template <>
+boost::shared_ptr<Region> make<Ellipse>(StringVector const& v) {
+    return lsst::qserv::qproc::getEllipseFromParams(convertVec<Coordinate>(v));
+}
+template <>
+boost::shared_ptr<Region> make<ConvexPolygon>(StringVector const& v) {
+    return lsst::qserv::qproc::getConvexPolyFromParams(convertVec<Coordinate>(v));
+}
+
+typedef boost::shared_ptr<Region>(*MakeFunc)(StringVector const& v);
+
+struct FuncMap {
+    FuncMap() {
+        fMap["box"] = make<Box>;
+        fMap["circle"] = make<Circle>;
+        fMap["ellipse"] = make<Ellipse>;
+        fMap["poly"] = make<ConvexPolygon>;
+        fMap["qserv_areaspec_box"] = make<Box>;
+        fMap["qserv_areaspec_circle"] = make<Circle>;
+        fMap["qserv_areaspec_ellipse"] = make<Ellipse>;
+        fMap["qserv_areaspec_poly"] = make<ConvexPolygon>;
+    }
+    typedef std::map<std::string, MakeFunc> Map;
+    Map fMap;
+};
+
+static FuncMap funcMap;
+} // anonymous namespace
+
+namespace lsst {
+namespace qserv {
+namespace qproc {
+typedef std::vector<boost::shared_ptr<Region> > RegionPtrVector;
+
+boost::shared_ptr<Region> getRegion(query::Constraint const& c) {
+    FuncMap::Map::const_iterator i = funcMap.fMap.find(c.name);
+    if(i != funcMap.fMap.end()) {
+        return i->second(c.params);
+    }
+    return boost::shared_ptr<Region>();
+}
+
+ChunkSpec convertSgSubChunks(SubChunks const& sc) {
+    ChunkSpec cs;
+    cs.chunkId = sc.chunkId;
+    cs.subChunks.resize(sc.subChunkIds.size());
+    std::copy(sc.subChunkIds.begin(), sc.subChunkIds.end(),
+              cs.subChunks.begin());
+    return cs;
+}
+
+bool isIndex(query::Constraint const& c) {
+    return c.name == "sIndex";
+}
```

Comment:
I reflowed the logic as the later comment suggested, so the code doesn't need these any more, so things should be fine.